File: 128_Longest_Consecutive_Sequence.go
package main

import (
	"fmt"
)

func longestConsecutive(nums []int) int {
	hashset := make(map[int]bool)
	for _, num := range nums {
		hashset[num] = true
	}

	longestSeq := 0

	for num := range hashset {
		// Check if this is the start of a sequence
		if !hashset[num-1] {
			seq := 1
			current := num + 1
			for hashset[current] {
				seq++
				current++
			}
			if seq > longestSeq {
				longestSeq = seq
			}
		}
	}

	return longestSeq
}

func main() {
	inputNums := []int{0, 3, 7, 2, 5, 8, 4, 6, 0, 1}
	res := longestConsecutive(inputNums)
	fmt.Println(res) // Output: 9
}

File: 128_Longest_Consecutive_Sequence.py
from typing import List
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        
        longest_seq = 0
        hashset = set(nums)
        for num in hashset:

            #determine whether the num is the entry point of a sequence
            if (num - 1) not in hashset:
                seq = 1
                while (num + seq) in hashset:
                    seq += 1
                longest_seq = max(seq, longest_seq)
        return longest_seq


input = [0,3,7,2,5,8,4,6,0,1]
solution = Solution()
res = solution.longestConsecutive(input)
print(res)